ddtrace/tracer.py

```diff
@@ -672,9 +672,26 @@ def _initialize_span_processors(self):
                 partial_flush_min_spans=self._partial_flush_min_spans,
                 trace_processors=trace_processors,
                 writer=self.writer,
-            ),
+            )
         ]  # type: List[SpanProcessor]
 
+        if appsec_enabled:
+            try:
+                from .appsec.processor import AppSecSpanProcessor
+
+                appsec_span_processor = AppSecSpanProcessor()
+                self._span_processors.append(appsec_span_processor)
+            except Exception as e:
```

Comment:
I think we should have this hard-raise since the user has to opt in to appsec. If they opt in to appsec and it breaks then it should be a loud failure.

I'd only prefer logging here if there were no way to avoid the error but the user can just disable appsec in this case.

Reply:
That's a good point. Historically, we want this not to fail as in dev environments it can frustrates the user. @stombre do you know what was the call on other libs? I don't remember.

Reply:
As of the [AppSec spec](https://datadoghq.atlassian.net/wiki/spaces/APS/pages/2097317007/Start-up+Rules+loading) we need to just disable appsec. This is what's implemented in the other libraries. It might be worth considering making that configurable however! I will add that to the discussions for the next milestone of the team.